src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -796,6 +801,10 @@ def load_arguments(self, _):
             if self.supported_api_version(min_api='2017-08-01', resource_type=ResourceType.MGMT_NETWORK):
                 PublicIPAddressSkuName = self.get_models('PublicIPAddressSkuName', resource_type=ResourceType.MGMT_NETWORK)
                 c.argument('public_ip_sku', help='Public IP SKU. It is set to Basic by default.', default=None, arg_type=get_enum_type(PublicIPAddressSkuName))
+            c.argument('nic_delete_option', nargs='+', min_api='2021-03-01',
+                       help='Specify what happens to the network interface when the VM is deleted. Use a singular '
+                       'value to apply on all resources, or use <Name>=<Value> to configure '
+                       'the delete behavior for individual resources. Possible options are Delete and Detach.')
```

Comment:
> Possible options are Delete and Detach.

Is there an enumeration in the SDK that can represent possible values?

Reply:
No. As this parameter will accept one single string or multiple key=value pairs. For possible values for delete option, I add it in description.

Reply:
OK